### ISSUE TITLE
some .png image urls fail to display

### ISSUE THREAD

**Issue by CondensedMilk7** [TheWizardBear/terminal_images#4]
passing urls such as `terminal_images https://flagcdn.com/w320/ge.png` results in the following output:

```
▖▖▖▖▖▖▖▖▖▖▖▖▖▖▖▐▖▖▖▖▖▖▖▖▖▖▖▖▖▖▖▖▖▖▖▖▖▖▖▐▖▖▖▖▖▖▖▖▖▖▖▖▖▖▖
▖▖▖▖▖▘▝▖▖▖▖▖▖▖▖▐▖▖▖▖▀▖▀▖▖▖▖▖▖▖▖▗▀▖▀▖▖▖▖▐▖▖▖▖▖▖▖▗▘▝▀▖▖▖▖
▖▖▖▖▖▝▘▖▖▖▖▖▖▖▖▐▖▖▖▖▖▀▖▖▖▖▖▖▖▖▖▖▖▀▖▖▖▖▖▐▖▖▖▖▖▖▖▖▝▘▖▖▖▖▖
▀▀▀▀▀▀▀▀▀▀▀▀▖▖▖▝▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▘▖▖▖▀▀▀▀▀▀▀▀▀▀▀▀
▖▖▖▖▖▖▖▖▖▖▖▖▖▖▖▖▖▖▖▖▖▖▖▖▖▖▖▖▖▖▖▖▖▖▖▖▖▖▖▖▖▖▖▖▖▖▖▖▖▖▖▖▖▖▖
▀▀▀▀▀▀▀▀▀▀▀▀▖▖▖▗▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▖▖▖▖▀▀▀▀▀▀▀▀▀▀▀▀
▖▖▖▖▖▗▖▖▖▖▖▖▖▖▖▐▖▖▖▖▖▀▖▖▖▖▖▖▖▖▖▖▖▀▖▖▖▖▖▐▖▖▖▖▖▖▖▖▗▖▖▖▖▖▖
▖▖▖▖▘▖▗▘▖▖▖▖▖▖▖▐▖▖▖▖▀▖▀▘▖▖▖▖▖▖▖▝▀▖▀▖▖▖▖▐▖▖▖▖▖▖▖▝▖▗▀▖▖▖▖
▖▖▖▖▖▖▖▖▖▖▖▖▖▖▖▐▖▖▖▖▖▖▖▖▖▖▖▖▖▖▖▖▖▖▖▖▖▖▖▐▖▖▖▖▖▖▖▖▖▖▖▖▖▖▖
aN;NaN;NaNmaN;NaN;NaNm▖aN;NaN;NaNmaN;NaN;NaNm▖aN;NaN;NaNmaN;NaN;NaNm▖
aN;NaN;NaNmaN;NaN;NaNm▖aN;NaN;NaNmaN;NaN;NaNm▖aN;NaN;NaNmaN;NaN;NaNm▖
aN;NaN;NaNmaN;NaN;NaNm▖aN;NaN;NaNmaN;NaN;NaNm▖aN;NaN;NaNmaN;NaN;NaNm▖
aN;NaN;NaNmaN;NaN;NaNm▖aN;NaN;NaNmaN;NaN;NaNm▖aN;NaN;NaNmaN;NaN;NaNm▖
aN;NaN;NaNmaN;NaN;NaNm▖aN;NaN;NaNmaN;NaN;NaNm▖aN;NaN;NaNmaN;NaN;NaNm▖
aN;NaN;NaNmaN;NaN;NaNm▖aN;NaN;NaNmaN;NaN;NaNm▖aN;NaN;NaNmaN;NaN;NaNm▖
aN;NaN;NaNmaN;NaN;NaNm▖aN;NaN;NaNmaN;NaN;NaNm▖aN;NaN;NaNmaN;NaN;NaNm▖
aN;NaN;NaNmaN;NaN;NaNm▖aN;NaN;NaNmaN;NaN;NaNm▖aN;NaN;NaNmaN;NaN;NaNm▖
aN;NaN;NaNmaN;NaN;NaNm▖aN;NaN;NaNmaN;NaN;NaNm▖aN;NaN;NaNmaN;NaN;NaNm▖
```
The same endpoint with jpg however seems to work fine (such as https://flagcdn.com/w320/ge.jpg)

**Comment by mjrlowe** [comment]
Hmm, not sure what's causing this but I will try and look into it further.

**Comment by mjrlowe** [comment]
Fixed in version 3.1.0. The issue was that the PNG library I am using doesn't fully support PNGs with colour palettes (PLTE), so I have implemented the missing stuff myself.
